packages/backend/src/lib/git-instance.ts

```diff
@@ -299,6 +335,37 @@ export class GitInstance {
     }
   }
 
+  /**
+   * Checkout commit based on Git hash as ref
+   * to restore files from specific commit
+   * @param gitHash commit sha
+   * @param token personal token
+   */
+  async checkoutOnRef(gitHash: string, token: string): Promise<void> {
```

Comment:
I'm not using `git.clone` here because it's already done previously when doing `const gitInstance = await GitInstance.from(gitUrl, githubPersonalAccessToken!);` on the 'rollBackCommit' method